### PR TITLE
Feature/ball physics

### PR DESCRIPTION
**Pull request by RossTubes**
Ball physics updated, also added a death barrier and some scripts on all the pins and bounce objects. added mesh colliders so we can use the blockout model properly so the ball won't fall through it. also added some physics materials on the ball itself.

### CODE REVIEW

File: Assets/Scripts/physics/Ball.cs
Comment (DanielNijkamp) [inline Assets/Scripts/physics/Ball.cs:1]: Deze file mag weg als hij niet word gebruikt?

File: Assets/Scripts/physics/Bouncer.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Bouncer : MonoBehaviour
+{
+    public float explosionStrength = 100f;
```
Comment (DanielNijkamp) [inline Assets/Scripts/physics/Bouncer.cs]:
```suggestion
    [SerializeField] private float _explosionStrength;
```

File: Assets/Scripts/physics/Bouncer.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+    void OnCollisionEnter(Collision _other)
+    {
+      
+        _other.rigidbody.AddExplosionForce(explosionStrength, transform.position, 2f);
+        Debug.Log("test");
+    }
```
Comment (DanielNijkamp) [inline Assets/Scripts/physics/Bouncer.cs]:
```suggestion
    void OnCollisionEnter(Collision other)
    {
        other.rigidbody.AddExplosionForce(explosionStrength, transform.position, 2f);
    }
```

File: Assets/Scripts/physics/DeathBarrier.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DeathBarrier : MonoBehaviour
+{
+
+
```
Comment (DanielNijkamp) [inline Assets/Scripts/physics/DeathBarrier.cs]: empty space weghalen

File: Assets/Scripts/physics/DeathBarrier.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    public Transform respawnPosition; // Set this in the Unity editor to the desired respawn position
```
Comment (DanielNijkamp) [inline Assets/Scripts/physics/DeathBarrier.cs]:
```suggestion
    [SerializeField] private Transform _respawnPosition;
```

File: Assets/Scripts/physics/DeathBarrier.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+    private void OnCollisionEnter(Collision collision)
+    {
+        {
+            // Check if the colliding object is tagged as "Ball"
```
Comment (DanielNijkamp) [inline Assets/Scripts/physics/DeathBarrier.cs]:
```suggestion
```

File: Assets/Scripts/physics/DeathBarrier.cs [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+            if (collision.gameObject.CompareTag("Ball"))
+            {
+                // Move the ball to the respawn position
```
Comment (DanielNijkamp) [inline Assets/Scripts/physics/DeathBarrier.cs]:
```suggestion
```

**Review by DanielNijkamp — CHANGES_REQUESTED** [review]
Paar changes maar verder is wel ok qua code. fix ook nog documentatie in je wiki, UML + testfootage ook

File: Assets/Scripts/physics/Ball.cs
Comment (RossTubes) [reply Assets/Scripts/physics/Ball.cs:1]: aiit its gone

File: Assets/Scripts/physics/Bouncer.cs (hunk 1 above)
Comment (RossTubes) [reply Assets/Scripts/physics/Bouncer.cs]: changed

**Review by DanielNijkamp — APPROVED** [review]

**Pull request merged**